resources/qml/ExtruderIcon.qml

```diff
@@ -17,29 +17,26 @@ Item
     property alias textColor: extruderNumberText.color
     property bool extruderEnabled: true
 
-    UM.RecolorImage
+    Item
     {
-        id: mainIcon
+        opacity: extruderEnabled ? 1 : 0.4
         anchors.fill: parent
 
-        source: UM.Theme.getIcon("Extruder", "medium")
-        color: extruderEnabled ? materialColor: UM.Theme.getColor("disabled")
-    }
-
-    Rectangle
-    {
-        id: extruderNumberCircle
-
-        width: height
-        height: Math.round(parent.height / 2)
-        radius: Math.round(width)
-        color: UM.Theme.getColor("toolbar_background")
+        UM.RecolorImage
+        {
+            id: blorp
+            anchors.fill: parent
 
-        anchors
+            source: UM.Theme.getIcon("ExtruderColor", "medium")
+            color: materialColor
+        }
+        UM.RecolorImage
         {
-            horizontalCenter: parent.horizontalCenter
-            top: parent.top
-            topMargin: (parent.height - height) / 2
+            id: mainIcon
+            anchors.fill: parent
+
+            source: UM.Theme.getIcon("Extruder", "medium")
+            color: UM.Theme.getColor("text")
```

Comment:
My first thought was that this is not text and should not get the text colour assigned. But I understand that you'll want this to have the same colour as the text in the centre of the icon. Maybe this should just be linked to `extruderNumberText.color` then?